### ISSUE TITLE
White/Gray lines in input bars

### ISSUE THREAD

**Issue by skulblakka** [streamlink/streamlink-twitch-gui#357]
### Description
There are white lines in the text input bars like the search bar. They seem to appear and disappear somewhat randomly. 

### Reproduction steps
They don't always appear but I get them pretty often. For example when going to the "Streamlink" tab in the settings they will appear every time.

### Environment details (operating system, etc.)
Win10 Pro 64-bit Version 10.0.14393
NodeJS 7.2.1
npm 3.10.10
Tried with self compiled version (commit 412ca46) and the official build from the v.1.0.0 release

### Comments, logs, screenshots, etc.
![screen-settings](https://cloud.githubusercontent.com/assets/2007822/20981112/88d15c46-bcb3-11e6-8a3c-73493a4f7938.png)
![screen-follows](https://cloud.githubusercontent.com/assets/2007822/20981134/a09bc816-bcb3-11e6-9a81-76790307e88c.png)

While more noticeable with the dark theme enabled they also appear when set to default:
![screen-white](https://cloud.githubusercontent.com/assets/2007822/20981598/a9432d22-bcb5-11e6-9a87-dc52f6a471c6.png)
I edited the above screenshot a little bit so the lines are more noticeable:
![screen-white-edited](https://cloud.githubusercontent.com/assets/2007822/20981626/d1c699fa-bcb5-11e6-87cf-25e6f9373921.png)

**Comment by bastimeyer** [comment]
Looks like a weird rendering glitch of Chromium on Windows 10.
This doesn't happen in my Win10 VM. Which graphics drivers are you using? Maybe this is related.
Did you copy the old config/cache folder?

You could try running the debug build and then inspect one of the elements and disable some CSS properties/rules. The fields are just simple input elements with a border, nothing special. I'm not sure what could cause these glitches. I don't think it's a DOM (shadow-DOM) or CSS issue.

Maybe it's related to the text rendering issue of the past releases. This was caused by the very poor font rendering on Windows and semi-transparent text colors which were used since the implementation of the theme system. The new release has a fix for (almost) all selectors including text colors, but border colors are still semi-transparent. I'd be surprised, though, if Chromium is unable to render this correctly on Windows.

I have not found any reported bugs in the chromium bug tracker:
https://bugs.chromium.org/p/chromium/issues/list?can=2&q=os%3DWindows+component%3ABlink%3ELayout&colspec=ID+Pri+M+Stars+ReleaseBlock+Component+Status+Owner+Summary+OS+Modified&x=m&y=releaseblock&cells=ids

**Comment by skulblakka** [comment]
>Which graphics drivers are you using?

I am using the current NVidia driver (376.19). 

>Did you copy the old config/cache folder?

I didn't do anything with the cache/config folder. I guess the new version is using a different one because I had to redo my settings after the upgrade.

>You could try running the debug build and then inspect one of the elements and disable some CSS properties/rules.

I played around with it a little bit and when I disable ```border-color: rgba(255, 255, 255, 0.1575);``` (main.css line 2643) in dark theme and ```border-color: rgba(0, 0, 0, 0.1925);``` (main.css line 2621) in default theme the lines no longer appear.

**Comment by bastimeyer** [comment]
Do these glitches still appear if you change the border color to something without an alpha channel?

**Comment by skulblakka** [comment]
I just tried a bunch of different values and they seem to only appear if the alpha channel is between 0 and 1.

**Comment by bastimeyer** [comment]
Okay, thanks.
This can be fixed by the same method I'm using for the text colors.

**Comment by bastimeyer** [comment]
Could you please test this and validate that the glitches are gone? I don't have these glitches in my VM, as I've already said.

```bash
git fetch origin
git checkout origin/fix-input-border-glitch
grunt build
```

Thanks!

**Comment by skulblakka** [comment]
Can confirm. The glitches no longer appear.

**Comment by Poorchop** [comment]
I still have this problem after upgrading a moment ago. Do I have to completely uninstall the previous version first and delete my settings? I just installed over 1.0.0 and the graphical glitch persists.

**Comment by bastimeyer** [comment]
No, this has nothing to do with the settings, etc... The fix was just a change of css property values for various input element selectors.

Are there any other graphic glitches?
What is your configuration / environment?

Are you using multiple monitors?
Yesterday I have seen a video on YT from somebody who's giving a tutorial on how to install Streamlink Twitch GUI and he had glitches on his second/third monitor which I can't explain.

Those glitches should be reproducible in Chromium/Chrome `55.0.2883.75`.
It's better and easier to use a blank NW.js `v0.19.0` app, though, and then add DOM nodes (HTML elements) and add/toggle styles.

Could somebody who's seeing these glitches also run a debug build with a different NW.js version please?
https://nwjs.io/versions.json
This can be changed here:
https://github.com/streamlink/streamlink-twitch-gui/blob/v1.0.1/src/config/main.json#L4

Thanks

**Comment by skulblakka** [comment]
I can confirm that the issue does still exist and was not fixed in ae0e4da.

Here's the thing (and the reason why I though this was resolved): The glitches only appear in the release version. Running ```grunt build``` with ae0e4da or later the glitches will not appear.

I also tried building the latest version of the app (1f3bede) using different versions of NW.js (namely the latest one 0.19.2 and the previous version 0.18.8). While the issue persists with 0.19.2 it does not appear with 0.18.8.

**Comment by bastimeyer** [comment]
> The glitches only appear in the release version

Interesting...
Well, the only difference here is that the stylesheets are being minimized, but this shouldn't cause rendering issues.

I think this needs to be reported on the NW.js issue tracker if we can provide code that is able to reproduce the glitches.

**Comment by Poorchop** [comment]
>Are there any other graphic glitches?
None as far as I can tell.

>What is your configuration / environment?
Dark theme, Windows 8.1 64-bit, Nvidia driver

>Are you using multiple monitors?
Nope, just one monitor.

**Comment by ElPumpo** [comment]
Yea, I still wonder why people lie but that wasn't fixed - please remove from the v1.0.1 release notes.

win10 amd64 build 14986
gtx 970 - r376.33

**Comment by bastimeyer** [comment]
In software development stuff like this happens all the time, nobody is lying here. Please read the thread first before making such statements, thanks.
Changelogs of previous releases are not going to be modified.

**Comment by ElPumpo** [comment]
@bastimeyer woops didn't mean to release that comment, was going to say that wasn't fixed for me either, sorry if you feel offended :/

**Comment by dethzord** [comment]
May be related, but I get more graphical glitches beyond the white bars.  Screenshot below.

http://imgur.com/skJalhr

**Comment by bastimeyer** [comment]
I tested this on a natively running installation of Windows 10 and I did not get any graphics glitches.
This all looks like it's caused by GPU redraw/flush issues of Chromium. What's the output when opening `chrome://gpu` (in the sdk build)?

I would really appreciate it if someone could create an isolated NWjs demo app that reproduces this (instructions can be found on the NW.js website https://nwjs.io).

I think that it could be related to css animations/transitions, but I'm not sure.

**Comment by bastimeyer** [comment]
Even when using the normal build flavor? This is good news if it fixes the issue now.
I'm not upgrading to a beta release, though, so let's wait for a stable release, which will be every 6 weeks (NW.js is tied to the Chromium release cycle).

**Comment by skulblakka** [comment]
>Even when using the normal build flavor? 

Yes.

**Comment by bastimeyer** [comment]
Chromium 56 has been released to the stable branch today. The NW.js devs have already reacted (very fast 👍) and released `0.20.0`.

@skulblakka, could you please see if the new NW.js release fixes the issue?
Set the version here:
https://github.com/streamlink/streamlink-twitch-gui/blob/v1.1.1/src/config/main.json#L4

Thanks a lot!

**Comment by skulblakka** [comment]
I just build the application with the new release and no longer experience any graphical glitches (I made sure to test the sdk build and the normal build this time).
